app/controllers/beta/projects_controller.rb

```diff
@@ -5,6 +5,12 @@ def show
     authorize current_user
     @project_id = params[:id]
     @project = current_user.projects.friendly.find @project_id
+    session[:current_team_slug] = @project
```

Comment:
Can we abstract this? Maybe you could move this piece of logic to the user model instead of repeating yourself.

Reply:
Right, I was thinking in create a method in user model like 

def team_from_project(project)
     project
       .teams
       .map { |team| team if teams.include? team }
       .compact
       .first
  end

and in projects controller call 
`session[:current_team_slug] = current_user.team_from_project(@project).slug`
what you think?